server/aws-lsp-identity/src/language-server/identityServer.ts

```diff
@@ -0,0 +1,31 @@
+import { createConnection, ProposedFeatures } from 'vscode-languageserver/node'
+import {
+    listProfilesRequestType,
+    ListProfilesParams,
+    ListProfilesResult,
+    ListProfilesError,
+} from '@aws/language-server-runtimes/protocol/identity-management'
+import { ProfileService } from './profiles/profileService'
+import {
+    UpdateProfileParams,
+    UpdateProfileResult,
+    UpdateProfileError,
+    updateProfileRequestType,
+} from '@aws/language-server-runtimes/protocol/identity-management'
+import { SharedConfigProfileStore } from './profiles/sharedConfigProfileStore'
+
+const connection = createConnection(ProposedFeatures.all)
+
+const profileServer = new ProfileService(new SharedConfigProfileStore())
+
+connection.onRequest<ListProfilesParams, ListProfilesResult, void, ListProfilesError, void>(
+    listProfilesRequestType,
+    profileServer.listProfiles.bind(profileServer)
+)
+
+connection.onRequest<UpdateProfileParams, UpdateProfileResult, void, UpdateProfileError, void>(
+    updateProfileRequestType,
+    profileServer.updateProfile.bind(profileServer)
+)
+
+connection.listen()
```

Comment:
Why is this server is implemented as separate connection listener? This essentially makes it a separate standalone language server implementation, which can't be bundled with Runtimes Server implementation following Runtimes framework.

Standalone Runtime implementation exposes `identityManagement` feature and manages LSP connection, so this server does not have to work on raw LSP connection. https://github.com/aws/language-server-runtimes/blob/main/runtimes/runtimes/standalone.ts#L254-L264

Also this implementation does not allow to bundle this IdentityServer with other Servers in one bundle and will always require separate file.

Reply:
Yep, I realized this last night as I started implementing and testing SsoTokenChanged.  I will pull over the updates from that PR for IdentityServer and related files to this one.

Reply:
got it :)